src/hyperlink/_url.py

```diff
@@ -1705,59 +1822,85 @@ def child(self, *segments):
         new_segs = [_encode_reserved(s) for s in segments]
         return self.__class__(self._url.child(*new_segs))
 
-    def normalize(self, *a, **kw):
+    def normalize(self, scheme=True, host=True, path=True, query=True,
+                  fragment=True, userinfo=True, percents=True):
+        # type: (bool, bool, bool, bool, bool, bool, bool) -> DecodedURL
         """Return a new `DecodedURL` wrapping the result of
         :meth:`~hyperlink.URL.normalize()`
         """
-        return self.__class__(self._url.normalize(*a, **kw))
+        return self.__class__(self._url.normalize(
+            scheme, host, path, query, fragment, userinfo, percents
+        ))
 
     @property
     def absolute(self):
+        # type: () -> bool
         return self._url.absolute
 
     @property
     def scheme(self):
+        # type: () -> Text
         return self._url.scheme
 
     @property
     def host(self):
+        # type: () -> Text
         return _decode_host(self._url.host)
 
     @property
     def port(self):
+        # type: () -> Optional[int]
         return self._url.port
 
     @property
     def rooted(self):
+        # type: () -> bool
         return self._url.rooted
 
     @property
     def path(self):
+        # type: () -> Sequence[Text]
         try:
-            return self._path
+            return cast(
+                Tuple[Text, ...],
+                self._path  # type: ignore[has-type] can't determine
+            )
```

Comment:
Not sure why `mypy` can't figure out the type of `self._path` here, especially given that I'm telling it with the `cast` here.
There are a few more of these below.